src/Microsoft.DocAsCode.MarkdigEngine.Extensions/Inclusion/InclusionContext.cs

```diff
@@ -89,9 +110,11 @@ public static bool IsCircularReference(object file, out IEnumerable<object> depe
         {
             dependencyChain = null;
 
-            if (t_files.Contains(file))
+            var markupStack = t_markupStacks.Value;
+            var inclusionStack = markupStack.Count > 0 ? markupStack.Peek().inclusionStack : null;
+            if (inclusionStack.Contains(file))

```

Comment:
Can inclusionStack be null?

Reply:
nop, it is always initialized to an empty stack.

Reply:
oops